anesthetic/plot.py

```diff
@@ -39,6 +40,16 @@
 class AxesSeries(pandas.Series):
     """Anesthetic's axes version of `pandas.Series`."""
 
+    def __init__(self, data=None, index=None, fig=None, ncol=None, labels=None,
+                 gridspec_kw=None, subplot_spec=None, *args, **kwargs):
+        if data is None and index is not None:
+            data = pandas.Series(np.full(np.shape(index), None), index=index)
+            data = self.axes_series(data=data, fig=fig, ncol=ncol,
+                                    gridspec_kw=gridspec_kw,
+                                    subplot_spec=subplot_spec)
+            self._set_xlabels(axes=data, labels=labels)
+        super().__init__(data=data, index=index, *args, **kwargs)
```

Comment:
This is actually more in keeping with the other pandas extensions we have (`WeightedDataFrame` etc), in comparison with the `@classmethod` I had in mind.